Sources/PostgresNIO/New/Messages/Bind.swift

```diff
@@ -18,10 +18,12 @@ extension PSQLFrontendMessage {
             // zero to indicate that there are no parameters or that the parameters all use the
             // default format (text); or one, in which case the specified format code is applied
             // to all parameters; or it can equal the actual number of parameters.
-            buffer.writeInteger(1, as: Int16.self)
+            buffer.writeInteger(Int16(self.parameters.count))
             
             // The parameter format codes. Each must presently be zero (text) or one (binary).
-            buffer.writeInteger(1, as: Int16.self)
+            self.parameters.forEach {
+                buffer.writeInteger($0.psqlFormat.rawValue)
+            }
```

Comment:
Should reserve capacity in the buffer to minimise intermediate allocs?

Reply:
Although we have to encode the actual parameters as well which make it harder to know how much capacity to reserve.

Reply:
Maybe in another pr at a later point?

Reply:
Fine by me.